packages/dataviews/src/filter-and-sort-data-view.ts

```diff
@@ -143,6 +143,15 @@ export function filterSortAndPaginate< Item >(
 				const valueA = fieldToSort.getValue( { item: a } ) ?? '';
 				const valueB = fieldToSort.getValue( { item: b } ) ?? '';
 
+				if ( fieldToSort.type === 'integer' ) {
+					// TODO: should we pass the item instead of the resolved value?
+					return fieldToSort.sort(
+						valueA,
+						valueB,
+						view.sort?.direction ?? 'desc'
+					);
+				}
+
 				if (
 					typeof valueA === 'number' &&
```

Comment:
Note that I've left the existing logic here because `type` is optional. When/If make it required this could be removed.